### PR TITLE
API-3309 - Cache session index

### PR DESCRIPTION
**Pull request by bdgrider**
https://vajira.max.gov/browse/API-3309

### CODE REVIEW

File: saml-proxy/docker-compose.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
 version: '2.1'
 services:
+  redis:
+    image: redis
+    networks: 
+      - 'default'
+    expose:
```
Comment (crolarlibertyva) [inline saml-proxy/docker-compose.yml:7]: port mapping localhost eg
`    ports:
      - 6379:6379
`

**Review by crolarlibertyva — CHANGES_REQUESTED** [review]

**Review by crolarlibertyva — APPROVED** [review]

File: saml-proxy/src/routes/index.js [hunk 2]
```diff
@@ -168,7 +170,9 @@ export default function configureExpress(
     }
   });
 
-  addRoutes(app, idpOptions, spOptions);
+  /* exported variableName */
+  //const cache = new RedisCache();
```
Comment (esubocz-va) [inline saml-proxy/src/routes/index.js:174]: this comment can probably be taken out?

**Review by esubocz-va — APPROVED** [review]

File: saml-proxy/src/routes/acsHandlers.ts [hunk 3]
```diff
@@ -168,6 +168,45 @@ export const testLevelOfAssuranceOrRedirect = (
   }
 };
 
+export const validateIdpResponse = (cache: ICache) => {
+  return async (req: IConfiguredRequest, res: Response, next: NextFunction) => {
+    const sessionIndex = req?.user?.authnContext?.sessionIndex;
```
Comment (dbrown-va) [inline saml-proxy/src/routes/acsHandlers.ts:173]: Out of an abundance of caution, I would lean towards adding a flag for us to be able to enable solely in dev for a couple days to ensure expected behavior prior to enabling in higher environments.

We will also need a corresponding devops PR anyway once the cache set up is in place.

**Review by dbrown-va — APPROVED** [review]
Confirmed locally with the `cacheEnabled` behavior both true & false.

Devops PR already merged and only enables dev & staging at this time.

**Pull request merged**